xDrip/Scenes/Entries/EntriesList/EntriesListInteractor.swift

```diff
@@ -24,10 +27,47 @@ final class EntriesListInteractor: EntriesListBusinessLogic, EntriesListDataStor
     var presenter: EntriesListPresentationLogic?
     var router: EntriesListRoutingLogic?
     
+    private var entries: [AbstractEntry] = []
```

Comment:
While interactor is responsible for business logic, it should delegate most of his work to workers. Also, a more abstract approach is required here.

Do:
1. Create abstract
```swift
protocol EntriesListEntryPersistenceWorker {
  func fetchEntries() -> [AbstractEntry]
  func deleteEntry(_ entry: AbstractEntry)
}
```
2. Create concrete implementations for different types of entries, e.g.
```swift
final class EntriesListCarbsPersistenceWorker: EntriesListEntryPersistenceWorker {
  // Dummy for now
  private var carbs: [CarbEntry] = [/* fill with dummy values */]
  func fetchEntries() -> [AbstractEntry] {
    return carbs
  }

  func deleteEntry(_ entry: AbstractEntry) {
    carbs.remove()...
  }
}
```
3. Your interactor should know only about `EntriesListEntryPersistenceWorker` protocol. It shouldn't know neither about concrete implementations nor about objects array.
4. Those workers should be injected during the scene creation process. Create a method in view controller
```swift
func injectDependencies(interactorPersistenceWorker: EntriesListEntryPersistenceWorker)
```
5. Use that worker's methods in the interactor